Shared/Samples/Play KML tour/PlayKMLTourView.swift

```diff
@@ -0,0 +1,140 @@
+    @State private var error: Error? {
+        didSet { isShowingErrorAlert = error != nil }
+    }
+    
+    /// A Boolean value that indicates whether the KML tour is not ready to be played.
+    var tourIsNotReady: Bool {
+        tourStatus == .notInitialized || tourStatus == .initializing
+    }
+    
+    var body: some View {
+        // Create a scene view with a scene and a viewpoint.
+        SceneView(scene: scene, viewpoint: viewpoint)
+            .onViewpointChanged(kind: .centerAndScale) { viewpoint = $0 }
+            .task {
+                do {
+                    // Add KML layer to the scene.
+                    let dataset = KMLDataset(url: .esriTour)
+                    try await dataset.load()
+                    let layer = KMLLayer(dataset: dataset)
+                    scene.addOperationalLayer(layer)
+                    
+                    // Set the tour controller with first tour in the dataset.
+                    if let tour = firstKMLTour(in: dataset) {
+                        tourController.tour = tour
+                        
+                        // Listen for tour status updates.
+                        for await status in tourController.tour!.$status {
+                            tourStatus = status
+                        }
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    ZStack {
+                        HStack {
+                            Button {
+                                tourController.reset()
+                                viewpoint = scene.initialViewpoint
+                            } label: {
+                                Image(systemName: "gobackward")
+                            }
+                            .disabled(tourIsNotReady || tourStatus == .initialized)
+                            Spacer()
+                        }
+                        
+                        Button {
+                            tourStatus == .playing ? tourController.pause() : tourController.play()
+                        } label: {
+                            Image(systemName: tourStatus == .playing ? "pause.fill" : "play.fill")
+                        }
+                        .disabled(tourIsNotReady)
+                    }
+                }
+            }
+            .alert(isPresented: $isShowingErrorAlert, presentingError: error)
+    }
+}
+
+private extension PlayKMLTourView {
+    /// Finds the first KML tour in a KML dataset.
+    /// - Parameter dataset: The KML dataset to search through.
+    /// - Returns: The first `KMLTour` object in the dataset if any.
+    private func firstKMLTour(in dataset: KMLDataset) -> KMLTour? {
+        var nodes = dataset.rootNodes
+        var i = 0
+        
+        // Loop through the nodes until a tour is found.
+        while i < nodes.count {
```

Comment:
Can the while loop and i counter variable be replaced with a `forEach { node in ... }` loop for improved readability?

Reply:
Okay, I would prefer whichever method is more readable. If the while loop seems more readable to you then keep the code as-is.

Reply:
Sounds good. I'll let @mhdostal comment his opinion on this.